src/java.base/share/classes/sun/security/ssl/CipherSuite.java

```diff
@@ -894,39 +904,24 @@ private CipherSuite(int id, boolean isDefaultEnabled,
     }
 
     static CipherSuite nameOf(String ciperSuiteName) {
-        for (CipherSuite cs : CipherSuite.values()) {
-            if (cs.name.equals(ciperSuiteName) ||
-                    cs.aliases.contains(ciperSuiteName)) {
-                return cs;
-            }
-        }
-
-        return null;
+        return maps_name.get(ciperSuiteName);
     }
 
     static CipherSuite valueOf(int id) {
-        for (CipherSuite cs : CipherSuite.values()) {
-            if (cs.id == id) {
-                return cs;
-            }
-        }
-
-        return null;
+        return maps_id.get(id);
     }
 
     static String nameOf(int id) {
-        for (CipherSuite cs : CipherSuite.values()) {
-            if (cs.id == id) {
-                return cs.name;
-            }
+        if (maps_id.containsKey(id)) {
+            return maps_id.get(id).name;
```

Comment:
Would it make sense to skip `containsKey` and null-check the value returned by `get` instead?

Reply:
That's what containsKey() does, returning false of the value is null.

Reply:
Right. But calling `containsKey` and then `get` results in two hash map lookups. If we cache the result of `get`, we only need one lookup, which should be a tiny bit faster. See https://stackoverflow.com/q/14601016

Reply:
I feel that by checking for null, we can reduce one lookup at the Hashtable. Will update the pull request.

Reply:
Updated in the new revision.